## BUSINESS UNDERSTANDING

## 1.1 Introduction

The global marketplace has witnessed everchanging trends in different industries, and the real estate sector is no exception.To remain relevant and increase profitability in the Real estate market it is advantageous for leading Real Estate companies to study this trends so as to stay ahead. Happy Homes Partners are investors in the real estate market and have acknowledged that data science plays a pivotal role by uncovering hidden investment opportunities and steering clients towards profitable ventures.

## 1.2 Business Problem

Happy Homes Partners is looking to expand their property investments and are looking for optimal locations for their investment. We as Tech to Tech hub have a primary objective to facilitate informed real estate investment decisions for the Happy Homes Partners. Using the extensive housing dataset from Zillow, covering the period from April 1996 to April 2018, the project seeks to pinpoint the top 5 zip codes with the most promising investment potential.

## 1.3 Objectives

1. To identify the top 5 best zipcode for Happy Homes Partners to consider?

2. To identify the top 5 zip codes with the highest ROI.

3. To develop time series models to forecast real estate prices for different zip codes over various time horizons.

4. To establish cities that are optimal for both short-term and long-term investment

## DATA UNDERSTANDING

We will be using the Zillow Housing dataset, a CSV file for this project.
The dataset covers a large period from April 1996 to April 2018, allowing us to study long-term trends and different market conditions. It's a valuable resource for analyzing how prices have changed over time and predicting future trends.

## 2.1 Summary of the Dataset

RegionID - This is unique Id for the Regions.

SizeRank - This is the ranking done based on the size of the Region.

RegionName - This field contains the zip code of the Region.

RegionType - Type of Region is Zip.

StateName - State.

City - This column provide the specific City Name of Housing Data.

Metro - This provide the name of the metro city around that Region.

County Name - This is the County Name for that Region.

Months Column - These Columns contains the prices of Region for every month.

## 2.2 Data Inspection

In [28]:
# import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [29]:
# load the datasets
df = pd.read_csv('zillow_data.csv')

# View the first 5 rows
df.head()


RegionID  RegionName      City State              Metro CountyName  \
0     84654       60657   Chicago    IL            Chicago       Cook   
1     90668       75070  McKinney    TX  Dallas-Fort Worth     Collin   
2     91982       77494      Katy    TX            Houston     Harris   
3     84616       60614   Chicago    IL            Chicago       Cook   
4     93144       79936   El Paso    TX            El Paso    El Paso   

   SizeRank   1996-04   1996-05   1996-06   1996-07   1996-08   1996-09  \
0         1  334200.0  335400.0  336500.0  337600.0  338500.0  339500.0   
1         2  235700.0  236900.0  236700.0  235400.0  233300.0  230600.0   
2         3  210400.0  212200.0  212200.0  210700.0  208300.0  205500.0   
3         4  498100.0  500900.0  503100.0  504600.0  505500.0  505700.0   
4         5   77300.0   77300.0   77300.0   77300.0   77400.0   77500.0   

    1996-10   1996-11   1996-12   1997-01   1997-02   1997-03   1997-04  \
0  340400.0  341300.0  342600.0  344400.0  345700.0  346700.0  347800.0   
1  227300.0  223400.0  219600.0  215800.0  211100.0  205700.0  200900.0   
2  202500.0  199800.0  198300.0  197300.0  195400.0  193000.0  191800.0   
3  505300.0  504200.0  503600.0  503400.0  502200.0  500000.0  497900.0   
4   77600.0   77700.0   77700.0   77800.0   77900.0   77900.0   77800.0   

    1997-05   1997-06   1997-07   1997-08   1997-09   1997-10   1997-11  \
0  349000.0  350400.0  352000.0  353900.0  356200.0  358800.0  361800.0   
1  196800.0  193600.0  191400.0  190400.0  190800.0  192700.0  196000.0   
2  191800.0  193000.0  195200.0  198400.0  202800.0  208000.0  213800.0   
3  496300.0  495200.0  494700.0  494900.0  496200.0  498600.0  502000.0   
4   77800.0   77800.0   77800.0   77800.0   77900.0   78100.0   78200.0   

    1997-12   1998-01   1998-02   1998-03   1998-04   1998-05   1998-06  \
0  365700.0  370200.0  374700.0  378900.0  383500.0  388300.0  393300.0   
1  201300.0  207400.0  212200.0  214600.0  215100.0  213400.0  210200.0   
2  220700.0  227500.0  231800.0  233400.0  233900.0  233500.0  233300.0   
3  507600.0  514900.0  522200.0  529500.0  537900.0  546900.0  556400.0   
4   78400.0   78600.0   78800.0   79000.0   79100.0   79200.0   79300.0   

    1998-07   1998-08   1998-09   1998-10   1998-11   1998-12   1999-01  \
0  398500.0  403800.0  409100.0  414600.0  420100.0  426200.0  432600.0   
1  206100.0  202100.0  198800.0  196100.0  194100.0  193400.0  193400.0   
2  234300.0  237400.0  242800.0  250200.0  258600.0  268000.0  277000.0   
3  566100.0  575600.0  584800.0  593500.0  601600.0  610100.0  618600.0   
4   79300.0   79300.0   79400.0   79500.0   79500.0   79600.0   79700.0   

    1999-02   1999-03   1999-04   1999-05   1999-06   1999-07   1999-08  \
0  438600.0  444200.0  450000.0  455900.0  462100.0  468500.0  475300.0   
1  193100.0  192700.0  193000.0  193700.0  194800.0  196100.0  197800.0   
2  283600.0  288500.0  293900.0  299200.0  304300.0  308600.0  311400.0   
3  625600.0  631100.0  636600.0  642100.0  647600.0  653300.0  659300.0   
4   79900.0   80100.0   80300.0   80600.0   80900.0   81200.0   81400.0   

    1999-09   1999-10  ...    2014-03    2014-04    2014-05    2014-06  \
0  482500.0  490200.0  ...   863900.0   872900.0   883300.0   889500.0   
1  199700.0  201900.0  ...   234200.0   235400.0   236600.0   238500.0   
2  312300.0  311900.0  ...   282100.0   284200.0   286000.0   288300.0   
3  665800.0  672900.0  ...  1149900.0  1155200.0  1160100.0  1163300.0   
4   81700.0   82100.0  ...   112000.0   112500.0   112700.0   113100.0   

   2014-07  2014-08  2014-09  2014-10  2014-11  2014-12  2015-01  2015-02  \
0   892800   893600   891300   889900   891500   893000   893000   895000   
1   240500   242600   244700   246300   247600   249600   251400   253000   
2   290700   293300   295900   298300   300200   301300   301700   302400   
3  1167700  1173900  1175100  1173500  1175500  1178500  1176400  1174600   
4   113900   114400   114500

In [30]:
# View last 5 rows
df.tail()

RegionID  RegionName                 City State            Metro  \
14718     58333        1338             Ashfield    MA  Greenfield Town   
14719     59107        3293            Woodstock    NH        Claremont   
14720     75672       40404                Berea    KY         Richmond   
14721     93733       81225  Mount Crested Butte    CO              NaN   
14722     95851       89155             Mesquite    NV        Las Vegas   

      CountyName  SizeRank   1996-04   1996-05   1996-06   1996-07   1996-08  \
14718   Franklin     14719   94600.0   94300.0   94000.0   93700.0   93400.0   
14719    Grafton     14720   92700.0   92500.0   92400.0   92200.0   92100.0   
14720    Madison     14721   57100.0   57300.0   57500.0   57700.0   58000.0   
14721   Gunnison     14722  191100.0  192400.0  193700.0  195000.0  196300.0   
14722      Clark     14723  176400.0  176300.0  176100.0  176000.0  175900.0   

        1996-09   1996-10   1996-11   1996-12   1997-01   1997-02   1997-03  \
14718   93200.0   93000.0   92900.0   92700.0   92600.0   92600.0   92600.0   
14719   91900.0   91700.0   91300.0   90900.0   90500.0   90100.0   89800.0   
14720   58200.0   58400.0   58700.0   59100.0   59500.0   59900.0   60300.0   
14721  197700.0  199100.0  200700.0  202600.0  204900.0  207100.0  209100.0   
14722  175800.0  175800.0  176000.0  176200.0  176500.0  176700.0  176800.0   

        1997-04   1997-05   1997-06   1997-07   1997-08   1997-09   1997-10  \
14718   92800.0   93100.0   93500.0   94000.0   94500.0   95100.0   95800.0   
14719   89600.0   89400.0   89300.0   89200.0   89100.0   88900.0   88700.0   
14720   60800.0   61300.0   62000.0   62600.0   63200.0   63800.0   64500.0   
14721  211200.0  213400.0  215800.0  218300.0  221000.0  223900.0  226900.0   
14722  176800.0  176900.0  177000.0  177100.0  177200.0  177400.0  177600.0   

        1997-11   1997-12   1998-01   1998-02   1998-03   1998-04   1998-05  \
14718   96500.0   97100.0   97800.0   98500.0   99200.0  100000.0  100800.0   
14719   88400.0   88100.0   87700.0   87400.0   87200.0   87100.0   87100.0   
14720   65200.0   65900.0   66700.0   67400.0   68200.0   69000.0   69800.0   
14721  230300.0  234300.0  238700.0  243000.0  247300.0  251800.0  256300.0   
14722  177900.0  178300.0  178900.0  179200.0  179400.0  179600.0  179700.0   

        1998-06   1998-07   1998-08   1998-09   1998-10   1998-11   1998-12  \
14718  101700.0  102600.0  103400.0  104200.0  104900.0  105600.0  106300.0   
14719   87100.0   87200.0   87300.0   87300.0   87400.0   87300.0   87300.0   
14720   70700.0   71500.0   72300.0   73000.0   73700.0   74300.0   74900.0   
14721  260900.0  265500.0  270100.0  274500.0  278900.0  283200.0  287900.0   
14722  179900.0  180100.0  180300.0  180600.0  180900.0  181300.0  181900.0   

        1999-01   1999-02   1999-03   1999-04   1999-05   1999-06   1999-07  \
14718  106800.0  107300.0  107800.0  108400.0  109000.0  109600.0  110200.0   
14719   87200.0   87300.0   87400.0   87600.0   87900.0   88300.0   88700.0   
14720   75500.0   76000.0   76400.0   76900.0   77300.0   77700.0   78000.0   
14721  292700.0  297100.0  301100.0  305000.0  308700.0  312400.0  315900.0   
14722  182500.0  182900.0  183200.0  183500.0  183900.0  184300.0  184700.0   

        1999-08   1999-09   1999-10  ...   2014-03   2014-04   2014-05  \
14718  110800.0  111400.0  112000.0  ...  187600.0  187300.0  187300.0   
14719   89200.0   89600.0   90000.0  ...  180000.0  180000.0  181400.0   
14720   78200.0   78400.0   78400.0  ...  105100.0  105200.0  104800.0   
14721  319200.0  322400.0  325500.0  ...  528900.0  532000.0  535200.0   
14722  185200.0  185800.0  186400.0  ...  266800.0  269200.0  271500.0   

        2014-06  2014-07  2014-08  2014-09  2014-10  2014-11  2014-12  \
14718  188100.0   189600   190800   189800   188500   189400   191100   
14719  181900.0   182200   183300   185800   187700   187500   184900   
14720  104200.0   104200   104700   

In [31]:
# Look at the shape
df.shape

(14723, 272)

Our data has 14723 rows and 272 columns. The data is in a wide Format, evident by having columns 8 onwards being with dates. 

In [32]:
# Looking at more information of the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14723 entries, 0 to 14722
Columns: 272 entries, RegionID to 2018-04
dtypes: float64(219), int64(49), object(4)
memory usage: 30.6+ MB


Our data has 14723 entries ranging from 0 to 14722 with 272 columns with 3 main data types
1. 219 columns with floating point numbers
2. 49 columns with interger data types
3. 4 columns with object data types